sdk/identity/identity/src/credentials/managedIdentityCredential/tokenExchangeMsi.ts

```diff
@@ -0,0 +1,116 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import fs from "fs";
+import { createHttpHeaders, PipelineRequestOptions } from "@azure/core-rest-pipeline";
+import { AccessToken, GetTokenOptions } from "@azure/core-auth";
+import { promisify } from "util";
+import { credentialLogger } from "../../util/logging";
+import { MSI, MSIConfiguration } from "./models";
+import { msiGenericGetToken } from "./utils";
+import { DefaultAuthorityHost } from "../../constants";
+
+const msiName = "ManagedIdentityCredential - Token Exchange";
+const logger = credentialLogger(msiName);
+
+const readFileAsync = promisify(fs.readFile);
+
+function expiresInParser(requestBody: any): number {
+  // Parses a string representation of the seconds since epoch into a number value
+  return Number(requestBody.expires_on);
+}
+
+function prepareRequestOptions(
+  scopes: string | string[],
+  clientAssertion: string,
+  clientId?: string
+): PipelineRequestOptions {
+  const queryParameters: any = {
```

Comment:
This is body data now, making the variable name misleading

Reply:
Ohh good point! thank you!